Documentation/github-connector.md

```diff
@@ -29,11 +29,20 @@ connectors:
     clientID: $GITHUB_CLIENT_ID
     clientSecret: $GITHUB_CLIENT_SECRET
     redirectURI: http://127.0.0.1:5556/dex/callback
-    # Optional organization to pull teams from, communicate through the
-    # "groups" scope.
+    # Optional organizations and teams, communicated through the
+    # "groups" scope. The 'org' field is kept for legacy users.
```

Comment:
Lots of "legacy" references. I'm starting to feel old :)

We can probably just drop these docs.